src/main/java/org/javarosa/xpath/expr/XPathBinaryOpExpr.java

```diff
@@ -48,6 +48,11 @@ public int hashCode() {
     public void readExternal(DataInputStream in, PrototypeFactory pf) throws IOException, DeserializationException {
         op = ExtUtil.readInt(in);
         readExpressions(in, pf);
+        computedCacheability = ExtUtil.readBool(in);
```

Comment:
these are repeated frequently, which is pretty dangerous for making these updates later. It seems like storing these flags (and the relevant serialization code) in a composed member object would be safer and save "do not repeat effort"

Reply:
@ctsims Do you mean something different here than by the similar comment right above?